### PR TITLE
Switch to local yum repository

### PR DESCRIPTION
**Pull request by midmarch**
1. Add role to initialize a local repository on the target machines, to support environments without access to any repositories.
2. Let other roles & tasks depend on RPM repositories, instead of copying their own dependency from the controller to the target machine

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,10 +17,17 @@ Which can be installed using:
...
 ```
 
-# Inventory
+## Inventory
 
 `anet_fqhn` should bet set to ANET's [fully qualified hostname](https://de.wikipedia.org/wiki/Fully-Qualified_Host_Name) corresponding with the way user shall access anet via their web browser.
 
+## Artifacts (dependencies)
+Artifacts should be installed on the Ansible controller in:
+- `{{ local_rpm_repo_folder }}` All none RPM artifacts (currently only keycloak distribution)
```
Comment (gjvoosten) [inline README.md:26]: `All non-RPM artifacts`

File: roles/jre8/tasks/main.yml [hunk 2]
```diff
@@ -3,27 +3,8 @@
...
-  copy:
-    src: '{{source_rpm_folder}}/{{jre_rpm}}'
-    dest: '{{remote_rpm_folder}}/{{jre_rpm}}'
-    remote_src: '{{remote_src}}'
-
 - name: Install Java JRE
   become: yes
   yum:
-    name: '{{remote_rpm_folder}}/{{jre_rpm}}'
+    name: 'jre1.8.x86_64'
```
Comment (gjvoosten) [inline roles/jre8/tasks/main.yml:9]: Do you need the architecture here? (you removed it on anet [ https://github.com/NCI-Agency/ansible.anet.playbook/pull/22/files#diff-f29a12a7fc026da07461dc94a7a75757a27213802511fb53e95d6289a56f8e36R5 ])
BTW JRE 1.8? Is that still needed when anet provides its own runtime? Or is that for Keycloak? In any case, we should probably switch to a supported (LTS) Java version.

File: roles/local-repo/tasks/install.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# Install createrepo on CentOS / RedHat 7.x
+
+- set_fact:
+    createrepo_rpms:
+      - 'createrepo-0.9.9-28.el7.noarch.rpm'
+      - 'python-deltarpm-3.6-3.el7.x86_64.rpm'
+      - 'deltarpm-3.6-3.el7.x86_64.rpm'
```
Comment (gjvoosten) [inline roles/local-repo/tasks/install.yml:7]: Put version numbers under the defaults?

File: roles/postgresql/defaults/main.yml [hunk 4]
```diff
@@ -1,7 +1,5 @@
 remote_rpm_folder: '{{remote_artifact_folder}}'
 postgresql_major_version: 12
```
Comment (gjvoosten) [inline roles/postgresql/defaults/main.yml:2]: Switch to PostgreSQL 13?

**Review by gjvoosten — CHANGES_REQUESTED** [review]

File: roles/jre8/tasks/main.yml (hunk 2 above)
Comment (midmarch) [reply roles/jre8/tasks/main.yml:9]: The JRE is required for Keycloak indeed.

File: roles/postgresql/defaults/main.yml (hunk 4 above)
Comment (midmarch) [reply roles/postgresql/defaults/main.yml:2]: Fine with me, but not this PR.

File: README.md [hunk 5]
```diff
@@ -17,10 +17,17 @@ Which can be installed using:
...
 ```
 
-# Inventory
+## Inventory
 
 `anet_fqhn` should bet set to ANET's [fully qualified hostname](https://de.wikipedia.org/wiki/Fully-Qualified_Host_Name) corresponding with the way user shall access anet via their web browser.
 
+## Artifacts (dependencies)
+Artifacts should be installed on the Ansible controller in:
+- `{{ local_rpm_repo_folder }}` All non-RPM artifacts (currently only keycloak distribution)
```
Comment (gjvoosten) [inline README.md:26]: `Keycloak` with a capital K?

File: roles/local-repo/tasks/transfer-repo.yml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+- name: Create Ansible RPM repository folder exists
```
Comment (gjvoosten) [inline roles/local-repo/tasks/transfer-repo.yml:1]: `Create Ansible RPM repository folder if it doesn't exist`?

**Review by gjvoosten — APPROVED** [review]

**Comment by midmarch** [comment]
Thank you so much for reviewing @gjvoosten !

**Pull request merged**